### ISSUE TITLE
[bug] compatibility for recipes using v2 python imports not working with conan 1.51.0

### ISSUE THREAD

**Issue by ChendoChap** [conan-io/conan#11813]
### Environment Details
  * Operating System+version: Ubuntu 22.04.1 LTS / Windows 11 21H2
  * Compiler+version: N/A
  * Conan version: 1.51.0
  * Python version: 3.10.4/3.10.6

### Steps to reproduce
1. try to install any package that has already updated its recipe for 2.0
eg: execute `conan install boost/1.79.0@`

### Notes
- The compatibility for `conan.errors` added by 1.47.0 doesn't seem to be working.
- I was able to reproduce this on multiple devices.

#### Ubuntu 22.04.1 LTS
```console
user@user-virtual-machine:~$ lsb_release -d
Description:	Ubuntu 22.04.1 LTS
user@user-virtual-machine:~$ python --version
Python 3.10.4
user@user-virtual-machine:~$ conan --version
Conan version 1.51.0
user@user-virtual-machine:~$ conan install boost/1.79.0@
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=11
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
 
boost/1.79.0: Not found in local cache, looking in remotes...
boost/1.79.0: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.79k]                                          
Downloading conanfile.py completed [81.33k]                                              
Downloading conan_export.tgz completed [1.50k]                                           
Decompressing conan_export.tgz completed [0.00k]                                         
boost/1.79.0: Downloaded recipe revision 0
ERROR: boost/1.79.0: Cannot load recipe.
Error loading conanfile at '/home/user/.conan/data/boost/1.79.0/_/_/export/conanfile.py': Unable to load conanfile in /home/user/.conan/data/boost/1.79.0/_/_/export/conanfile.py
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/user/.conan/data/boost/1.79.0/_/_/export/conanfile.py", line 8, in <module>
    from conan.errors import ConanException, ConanInvalidConfiguration
ModuleNotFoundError: No module named 'conan.errors'
 
user@user-virtual-machine:~$ 
```
#### Windows 11 21H2
```console
C:\Users\user>python --version
Python 3.10.6

C:\Users\user>conan --version
Conan version 1.51.0

C:\Users\user>conan install boost/1.79.0@
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=17
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

boost/1.79.0: Not found in local cache, looking in remotes...
boost/1.79.0: Trying with 'conancenter'...
Downloading conanmanifest.txt completed [0.79k]
Downloading conanfile.py completed [81.33k]
Downloading conan_export.tgz completed [1.50k]
Decompressing conan_export.tgz completed [0.00k]
boost/1.79.0: Downloaded recipe revision 0
ERROR: boost/1.79.0: Cannot load recipe.
Error loading conanfile at 'C:\Users\user\.conan\data\boost\1.79.0\_\_\export\conanfile.py': Unable to load conanfile in C:\Users\user\.conan\data\boost\1.79.0\_\_\export\conanfile.py
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\user\.conan\data\boost\1.79.0\_\_\export\conanfile.py", line 8, in <module>
    from conan.errors import ConanException, ConanInvalidConfiguration
ModuleNotFoundError: No module named 'conan.errors'
```

**Comment by jcar87** [comment]
Hi @ChendoChap, thanks for reporting this issue.

Could you confirm how Conan was installed? We have identified an issue with the pre-packaged binary installers (windows installer and Debian package) and the issue will be fixed in 1.51.1 which will be release shortly. If Conan is installed with pip (e.g. inside a virtual environment), this should work as expected even with the current release.

**Comment by ChendoChap** [comment]
That would explain it, I did use the debian package on the Ubuntu machine and on windows I installed it through chocolatey which presumably uses the binary installer in the background. Glad to hear it'll be fixed soon.

**Comment by czoido** [comment]
Closed by https://github.com/conan-io/conan/pull/11817 to be released very soon in 1.51.1

Thanks a lot for reporting!
